admin/spec/service/pupils-not-taking-check.service.spec.js

```diff
@@ -0,0 +1,130 @@
+'use strict'
+/* global describe beforeEach afterEach it expect */
+
+const sinon = require('sinon')
+require('sinon-mongoose')
+const proxyquire = require('proxyquire')
+const Pupil = require('../../models/pupil')
+const pupilNotTakingCheckService = require('../../services/pupils-not-taking-check.service')
+//const AttendanceCode = require('../../models/attendance-code')
+
+const attendanceCodesMock = require('../mocks/attendance-codes')
+const pupilsWithReasonsMock = require('../mocks/pupils-with-reason')
+const pupilsWithReasonsFormattedMock = require('../mocks/pupils-with-reason-formatted')
+
+/* global beforeEach, describe, it, expect */
+
+describe('Pupils are not taking the check. Service', () => {
+  let service
+  let sandbox
+
+  function setupService (mock) {
+    return proxyquire('../../services/pupils-not-taking-check.service.js', {
+      '../models/pupil': sandbox.mock(Pupil)
+      .expects('find')
+      .chain('sort')
+      .resolves(mock)
+    })
+  }
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create()
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  describe('fetchPupilsWithReasons successfully finds pupils', () => {
+    beforeEach(() => {
+      service = setupService(pupilsWithReasonsMock)
+    })
+
+    it('should return a list of pupils with reasons (happy path)', async (done) => {
+      try {
+        const pupilsWithReasons = await service.fetchPupilsWithReasons(9991001)
+        expect(pupilsWithReasons).toEqual(pupilsWithReasonsMock)
+        done()
+      } catch (error) {
+        done('Error found while testing fetchPupilsWithReasons')
```

Comment:
This style no longer works with jasmine.  Instead you have to do:
`expect(true).toBe(false)`
if you want to guard against it throwing.

Reply:
Fixed in next commit.